listener/iot-connection.ts

```diff
@@ -28,12 +29,22 @@ export async function getConnection(
     logError('No IoT endpoint could be found');
     throw new Error('IoT Endpoint address not found');
   }
+  const region = process.env['AWS_REGION'] || process.env['AWS_PROFILE_REGION'];
+  if (!region) {
+    logError('AWS_REGION or AWS_PROFILE_REGION not set');
+    throw new Error('AWS_REGION or AWS_PROFILE_REGION not set');
+  }
 
+  const provider = fromNodeProviderChain();
```

Comment:
I think using `fromNodeProviderChain` is the way to go. It works in a similar case I have on another project.